examples/voice-assistant/assistant_calls.py

```diff
@@ -0,0 +1,117 @@
+import asyncio
+import logging
+from enum import Enum
+from typing import Annotated
+
+from livekit import rtc
+from livekit.agents import (
+    JobContext,
+    JobRequest,
+    WorkerOptions,
+    cli,
+    llm,
+)
+from livekit.agents.voice_assistant import AssistantContext, VoiceAssistant
+from livekit.plugins import deepgram, elevenlabs, openai, silero
+
+
+class Room(Enum):
+    BEDROOM = "bedroom"
+    LIVING_ROOM = "living room"
+    KITCHEN = "kitchen"
+    BATHROOM = "bathroom"
+    OFFICE = "office"
+
+
+class AssistantFnc(llm.FunctionContext):
+    @llm.ai_callable(desc="Turn on/off the lights in a room")
+    async def toggle_light(
+        self,
+        room: Annotated[Room, llm.TypeInfo(desc="The specific room")],
+        status: bool,
+    ):
+        logging.info("toggle_light %s %s", room, status)
+        ctx = AssistantContext.get_current()
+        key = "enabled_rooms" if status else "disabled_rooms"
+        li = ctx.get_metadata(key, [])
+        li.append(room)
+        ctx.store_metadata(key, li)
+
+    @llm.ai_callable(desc="User want the assistant to stop/pause speaking")
+    def stop_speaking(self):
```

Comment:
When you ask the LLM to stop talking when telling a story, you don't want it to answer. Instead you just call this function and don't call Assistant.say

Reply:
I see, so what is the behavior of the assistant without this `stop_speaking` fn? Since we already support interruption at the agent level and we already are handling the interruption with `@assistant.on("agent_speech_interrupted")` is this fn needed?

BTW I'm asking because I don't know the answer and not because I'm recommending something different. But maybe since I'm confused others will be too.

Reply:
This is a very easy way to detect when to stop talking vs we interrupt to ask a new question

Reply:
When users are interrupting, the new speech is automatically sent to the LLM and we will get a new answer, this is how VoiceAssistant is working, wdyt?